### ISSUE TITLE
VS and CLI offline package cache have mismatching SHAs

### ISSUE THREAD

**Issue by muratg** [dotnet/cli#5897]
When dotnet is first executed after a CLI installation, dotnet uncompresses its offline package cache, and does a restore.  This restore creates a SHA. This SHA is different between dotnet initial installation and Visual Studio restore. 

![image](https://cloud.githubusercontent.com/assets/1808428/23521434/d925ea00-ff33-11e6-9017-565157fc3d6a.png)

**Comment by livarcocc** [comment]
@ericstj Would you know why the lzma decompressed nupkgs would differ for the CLI?

**Comment by ericstj** [comment]
Because we reconstruct the nupkg ZIPs from a list of files and the zip implementation is platform specific.    In @muratg's example above the files are the same but their compressed representation is different.

**Comment by ericstj** [comment]
FWIW I did try to get them byte for byte identical in v1 and I believe I had it working assuming the folks who originally packed the nupkg did it with the same zip implementation we used.  But all bets are off if someone used a different implementation (eg: pack with desktop, zip the folder using explorer or winzip, etc).  I'll get the order of entries correct but the zip headers might be different (since they are implementation specific), they may have used a different compression level, etc.
